### PR TITLE
Feat/signup

### PR DESCRIPTION
**Pull request by lisatsewang**
Feat/signup includes: sign in, sign up, sing out, add sign in button in nav bar, make sign out button actually sign people out

### CODE REVIEW

File: public/factories/factories.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      signup: signup,
+      signout: signout,
+      signin: signin
+    };
+
+    //////////////////////////////////////////////////////
+
+    function signup(user) {
+      return $http.post('/signup', user)
+        .then(function(res) {
```
Comment (CaptainKevin) [inline public/factories/factories.js]: separate anonymous function into a separate named function

Comment (CaptainKevin) [inline public/factories/factories.js]: Also for sign in and signout

File: server/signupLogin/signupLoginController.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  	.exec(function(err, data){
+  		if (err) { // if findOne err
+  			console.log("signup findOne error: ", err);
+  		} else { // if no findOne err
+  			if (!data) { // if no user found, save user
+          bcrypt.hash(req.body.password, null, null, function(err, hash) {
+            if (err) {
+              console.log("signup bcrypt err: ", err);
+            } else {
+      				var newUser = new Users({username: req.body.username, password: hash});
```
Comment (CaptainKevin) [inline server/signupLogin/signupLoginController.js]: Tab/Spacing issue - Need to use spaces, tabs get converted to four spaces

File: public/app.js [hunk 3]
```diff
@@ -1,6 +1,8 @@
 angular.module('app', [
   'app.navbar',
   'app.createEvent',
+  'app.signupLogin',
+  'app.factories',
```
Comment (MichaelGofron) [inline public/app.js:5]: We might want to just leave it in one factory until we refactor to events, users, and the eventsAndUsers factories

File: public/factories/factories.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
+(function() {
```
Comment (MichaelGofron) [inline public/factories/factories.js:1]: We will eventually refactor to a couple factories with users, events, and usersAndEvents

File: public/signupLogin/signupLoginController.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+  function signupLoginController($window, $location, Auth) {
+    
+      var vm = this;
+      vm.signup = signup;
+      vm.signin = signin;
+      vm.signout = Auth.signout;
+
+    function signup(user) {
+      Auth.signup(user)
+        .then(function(token) {
```
Comment (MichaelGofron) [inline public/signupLogin/signupLoginController.js]: anonymous function! I would recommend moving to a named function within the scope of signup()

File: server/signupLogin/signupLoginController.js [hunk 6]
```diff
@@ -0,0 +1,79 @@
+var Users = require('../users/usersModel');
+var bcrypt = require('bcrypt-nodejs');
+var jwt = require('jwt-simple');
+
+var signup = function(req, res) {
+	console.log("Post to signup: ", req.body);
+
+	// find user by username
+  Users.findOne({username: req.body.username})
+  	.exec(function(err, data){
```
Comment (MichaelGofron) [inline server/signupLogin/signupLoginController.js]: Change the anonymous function in exec here to named function

**Pull request merged**